frontend/controller/actions/identity.js

```diff
@@ -146,7 +146,20 @@ export default (sbp('sbp/selectors/register', {
           }
         }
       }
-      // note: leaving groups will happen when we sync the removeOurselves message
+      // NOTE: should sync all the identity contracts which are not part of same group
+      // but from the direct messages invited by another
+      const chatRoomUsers = uniq(Object.keys(
+        pickWhere(state.contracts, ({ type }) => type === 'gi.contracts/chatroom')
+      ).map(cID => Object.keys(state[cID].users)).flat())
+      const additionalIdentityContractIDs = await Promise.all(chatRoomUsers.filter(username => {
+        return getters.ourUsername !== username && !getters.ourContacts.includes(username)
+      }).map(username => sbp('namespace/lookup', username)))
```

Comment:
`sbp('namespace/lookup', username)` returns a Promise, so the resulting array will be full of promises, not strings.

Reply:
One way to address this would be to, below, instead of:

```js
await sbp('chelonia/contract/sync', identityContractID)
```

Do:

```js
// 'identityContractID' could be a promise
await sbp('chelonia/contract/sync', await identityContractID)
```

Reply:
@taoeffect, I think `additionalIdentityContractIDs` is the array of strings because `await` is used before Promise.all()

Reply:
Not sure how I missed that - you're right!